Create contributor-qualification.md (#15)
ethereum/kzg-ceremony-specs: contributorQualification.md

# Aims

> Our most important aim with the trusted setup is to avoid all practical possibility of a single actor controlling the contributions. A secondary aim is to avoid censoring anyone who wishes to contribute. 

Reviewer: ```suggestion
Our most important aim with the trusted setup is to avoid all practical possibility of a single actor knowing all the secrets. A secondary aim is to avoid censoring anyone who wishes to contribute. 
```

IMO this is an important distinction as not knowing a secret is a very different safety assumption than not controlling all the contributors

We will inevitably exclude some interested and genuine participants who don’t meet the criteria. The overriding aim is 
to avoid attempts by malicious operators controlling large numbers of sybil identities. A balanced approach will allow 
a large number of unique contributors and allow the possibility of a small number of sybil identities. 

# Contribution Flow

* Contributors will log in using one of the accepted authentication methods
* The chosen identifier will be validated. Only valid identifiers will be able to progress to the following steps
* A JWT will be issued in response to the login request. 
* The JWT will be required in request headers for all subsequent requests. The token will be valid for a period of several hours. The contributor should not need to go through login again before
completing their contribution, provided they do not log off or close the page.

# Authentication Methods

* Ethereum accounts 
* GitHub 

## Common Criteria

Identifiers must not have previously submitted a contribution to this ceremony, nor attempted a submission. 

## Authentication by ETH Account.
### Identifiers: Eth accounts and ENS names

> Participants will be identified using Ethereum accounts. We have no need to prohibit pseudonymous contributions, given that they have already passed anti-sybil checks.

Reviewer: ```suggestion
Participants can be identified using Ethereum accounts. We allow pseudonymous contributions, given that they have already passed anti-sybil checks.
```

However, well-known names may add to the credibility of the ceremony. For any address with an ENS name reverse-lookup configured, the ENS name will be used in place of the address. 

> Identifiers will be added to the transcript and will become part of the permanent record of the ceremony. 

Reviewer: Ok, this is currently not a part of the specs. I will need to modify the `transcript.json` to allow this.

### Acceptance Criteria

> The ceremony will entail a sign-in step for all contributors. The browser client will request that a message be signed, and will pass the address and signature to the server as a sign-in request.

Reviewer: ```suggestion
The ceremony will entail a sign-in step for all contributors. The contributor client will request that a message be signed, and will pass the address and signature to the server as a sign-in request.
```

The first attempt to sign in will trigger an assessment of the account’s eligibility.  
Eligible accounts will proceed to an ENS name lookup, and will then be registered by the server.

> To pass the assessment, the account will require a **nonce ≥ 3** on mainnet at the time the assessment is done.

Reviewer: De we not need to snapshot the nonces ahead of time to prevent extra registrations DoSing us?

Author: Perhaps we should. Then we could choose a block height around 1 April and use that.

## Authentication via GitHub

### Acceptance Criteria

GitHub accounts will be accepted if they have a commit dated prior to 1 August, 2022 00:00 UTC.

### Identifier
The GitHub handle will be used as the identifier in the ceremony contribution record. 

# Coordinator Handover

Handover of the coordinator role will require continued access to the list of all accounts used thus far in the ceremony.

> During hadnover, the retiring coordinator must export a list of all identifiers, and the new coordinator must import the list.

Reviewer: We should standardize this via a JSON schema
